packages/components-js/tests/e2e/puppeteer/specs/pin-code.e2e.ts

```diff
@@ -0,0 +1,618 @@
+import type { Page } from 'puppeteer';
+import {
+  addEventListener,
+  BASE_URL,
+  expectA11yToMatchSnapshot,
+  getAttribute,
+  getElementStyle,
+  getEventSummary,
+  getHTMLAttributes,
+  getLifecycleStatus,
+  getProperty,
+  goto,
+  selectNode,
+  setContentWithDesignSystem,
+  setProperty,
+  waitForStencilLifecycle,
+} from '../helpers';
+import { Components } from '@porsche-design-system/components';
+import { ElementHandle } from 'puppeteer';
+
+let page: Page;
+beforeEach(async () => {
+  page = await browser.newPage();
+  await browser
+    .defaultBrowserContext()
+    .overridePermissions(`${BASE_URL}/#`, ['clipboard-read', 'clipboard-write', 'clipboard-sanitized-write']);
+});
+afterEach(async () => await page.close());
+
+const getHost = () => selectNode(page, 'p-pin-code');
+const getLabel = () => selectNode(page, 'p-pin-code >>> .label__text');
+const getCurrentInput = () => selectNode(page, 'p-pin-code >>> #current-input');
+const getMessage = () => selectNode(page, 'p-pin-code >>> .message');
+const getHiddenInput = () => selectNode(page, 'p-pin-code input[slot="hidden-input"]');
+const getInput1 = () => selectNode(page, 'p-pin-code >>> input[aria-label="1-4"]');
```

Comment:
Perhaps selecting via `:nth-child` would be more future proof in case aria labels change?